sources/packages/api/src/msfaa-integration/msfaa-request.service.ts

```diff
@@ -0,0 +1,151 @@
+import { Injectable } from "@nestjs/common";
+import { InjectLogger } from "../common";
+import { MSFAANumber } from "../database/entities";
+import { LoggerService } from "../logger/logger.service";
+import { getUTCNow } from "../utilities";
+import { EntityManager } from "typeorm";
+import { MSFAANumberService, SequenceControlService } from "../services";
+import {
+  MSFAARecord,
+  MSFAAUploadResult,
+} from "./models/msfaa-integration.model";
+import { MSFAAIntegrationService } from "./msfaa-integration.service";
+
+@Injectable()
+export class MSFAARequestService {
+  constructor(
+    private readonly msfaaNumberService: MSFAANumberService,
+    private readonly msfaaService: MSFAAIntegrationService,
+    private readonly sequenceService: SequenceControlService,
+  ) {}
+
+  /**
+   * 1. Fetches the MSFAA records which are not sent for request.
+   * 2. Create Unique sequence for the request sent file.
+   * 3. Create the Request content for the MSFAA file by populating the
+   *      header, footer and trailer content.
+   * 4. Create the request filename with the file path for the MSFAA Request
+   *      sent File.
+   * 5. Upload the content to the zoneB SFTP server.
+   * 6. Update the MSFAA records, that are sent in the request sent file.
+   * @returns Processing MSFAA request result.
+   */
+  async processMSFAARequest(
+    offeringIntensity: string,
```

Comment:
We have the enum OfferingIntensity for this that is actually the one that is being used while calling this method.
Can you please change it to be the enum type?